File: back-end/src/main/java/es/ull/project/application/service/workflow/WorkFlowExportService.java
package es.ull.project.application.service.workflow;

import es.ull.project.domain.valueobject.ExportFile;
import es.ull.project.application.usecase.workflow.WorkFlowExportUseCase;
import es.ull.project.domain.entity.WorkFlow;
import org.springframework.stereotype.Service;
import java.util.Optional;

@Service
public class WorkFlowExportService implements WorkFlowExportUseCase {

    @Override
    public Optional<ExportFile> exportWorkFlow(WorkFlow workflow) {
        try {
            ExportFile exportFile = new ExportFile(workflow);
            return Optional.of(exportFile);
        } catch (Exception e) {
            return Optional.empty();
        }
    }
}

File: front-end/src/application/service/workflow/workflow-export-service.ts
import {  WorkFlowExportUseCase } from '../../usecase/workflow/workflow-export-use-case';
import type { WorkFlowRepository } from '../../repository/workflow-repository';
import { UllUUID, type Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { ExportFile } from '@/domain/value-object/export-file';

export class WorkFlowExportService implements WorkFlowExportUseCase {

  constructor(private readonly flowRepository: WorkFlowRepository) {}

  async exportWorkFlow(id: UllUUID): Promise<Either<DataError, ExportFile>> {
    return this.flowRepository.exportWorkFlow(id);
  }
}
